### DISCUSSION TITLE
Multiple camera error - [usb_cam_node_exe-2] terminate called after throwing an instance of 'char*'

### DISCUSSION THREAD

**Discussion by luisgrana2** [ros-drivers/usb_cam#284] [Q&A]
I am new working with two cameras using the usb_cam, and I keep encountering the same error: [usb_cam_node_exe-2] terminate called after throwing an instance of 'char*'. 
I add this to the launch camera.launch.py.
`CAMERAS = []
CAMERAS.append(
    CameraConfig(
        name='camera1',
        param_path=Path(USB_CAM_DIR, 'config', 'params_1.yaml')
    )
)

CAMERAS.append(
    CameraConfig(
        name='camera2',
        param_path=Path(USB_CAM_DIR, 'config', 'params_2.yaml')
    )
)`

and 

params_2.yaml is exactly the same as params_1, but I change the vid_device, in paramas_1.yaml I have /dev/video5.

`/**:
    ros__parameters:
      video_device: "/dev/video5"
      framerate: 30.0
      io_method: "mmap"
      namespace: "cam2"
      frame_id: "camera2"
      pixel_format: "yuyv"  # see usb_cam/supported_formats for list of supported formats
      image_width: 640
      image_height: 480
      camera_name: "camera2"
      camera_info_url: "package://usb_cam/config/camera_info.yaml"
      brightness: -1
      contrast: -1
      saturation: -1
      sharpness: -1
      gain: -1
      auto_white_balance: true
      white_balance: 4000
      autoexposure: true
      exposure: 100
      autofocus: false
      focus: -1`

Any idea of the error?

**Comment by flynneva** [comment]
@luisgrana2 this error comes from trying to open a device that either doesnt exist or isnt a v4l2 device (see issue #285). I've tried to improve the error reported with PR #286 👍🏼 sorry for the confusion!
